exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -97,97 +96,75 @@
+        .map(t -> t.getServiceId)
+        .collect(toList());
   }
 
-  private static TestKit newInstance(List<Class<? extends ServiceModule>> serviceModules,
+  private static TestKit newInstance(TestKitServiceInstances[] serviceInstances,
                                      EmulatedNodeType nodeType, short validatorCount,
-                                     @Nullable TimeProvider timeProvider) {
-    Injector frameworkInjector = Guice.createInjector(new TestKitFrameworkModule());
-    return newInstanceWithInjector(serviceModules, nodeType, validatorCount, timeProvider,
-        frameworkInjector);
-  }
-
-  private static TestKit newInstanceWithInjector(
-      List<Class<? extends ServiceModule>> serviceModules, EmulatedNodeType nodeType,
-      short validatorCount, @Nullable TimeProvider timeProvider, Injector frameworkInjector) {
-    Map<Short, UserServiceAdapter> serviceAdapters = toUserServiceAdapters(
-        serviceModules, frameworkInjector);
+                                     List<TimeServiceSpec> timeServiceSpecs) {
     boolean isAuditorNode = nodeType == EmulatedNodeType.AUDITOR;
-    UserServiceAdapter[] userServiceAdapters = serviceAdapters.values()
-        .toArray(new UserServiceAdapter[0]);
-    TimeProviderAdapter timeProviderAdapter =  timeProvider == null
-        ? null
-        : new TimeProviderAdapter(timeProvider);
-    long nativeHandle = nativeCreateTestKit(userServiceAdapters, isAuditorNode, validatorCount,
-        timeProviderAdapter);
-    return new TestKit(nativeHandle, serviceAdapters);
+    long nativeHandle = nativeCreateTestKit(serviceInstances, isAuditorNode, validatorCount,
+        timeServiceSpecs.toArray(new TimeServiceSpec[0]));
+    return new TestKit(nativeHandle, timeServiceSpecs);
   }
 
   /**
-   * Returns a list of user service adapters created from given service modules.
+   * Deploys and creates a single service with a single validator node in this TestKit network.
    */
-  private static Map<Short, UserServiceAdapter> toUserServiceAdapters(
-      List<Class<? extends ServiceModule>> serviceModules, Injector frameworkInjector) {
-    List<UserServiceAdapter> services = serviceModules.stream()
-        .map(s -> createUserServiceAdapter(s, frameworkInjector))
-        .collect(toList());
-    // UserServiceAdapters are removed, so this usage is a false signal for keeping getId()
-    return Maps.uniqueIndex(services, UserServiceAdapter::getId);
+  public static TestKit forService(ServiceArtifactId artifactId, String artifactFilename,
+                                   String serviceName, int serviceId, MessageLite configuration) {
+    checkServiceId(serviceId);
+    TestKitServiceInstances[] testKitServiceInstances = createTestKitSingleServiceInstance(
+        artifactId, artifactFilename, serviceName, serviceId, configuration);
+    return newInstance(
+        testKitServiceInstances, EmulatedNodeType.VALIDATOR, (short) 1, emptyList());
```

Comment:
I think using a builder here would reduce code duplication and simplify
things:

```java
    return new Builder()
        .withNodeType(EmulatedNodeType.VALIDATOR)
        .withDeployedArtifact(artifactId, artifactFilename)
        .withService(artifactId, serviceName, serviceId, configuration)
        .build();
```

createTestKitSingleServiceInstance would no longer be needed